src/register.jl

```diff
@@ -333,7 +337,14 @@ product_state(total::Int, bit_config::Integer; nbatch::Int=1) = product_state(Co
 product_state(::Type{T}, bit_str::BitStr; nbatch::Int=1) where T = ArrayReg{nbatch}(T, bit_str)
 
 function product_state(::Type{T}, total::Int, bit_config::Integer; nbatch::Int=1) where T
-    return ArrayReg{nbatch}(onehot(T, total, bit_config, nbatch))
+    if nbatch == 1
+        raw = onehot(T, total, bit_config, nbatch)
+    else
+        raw = zeros(T, nbatch, 1<<total)
+        raw[:,Int(bit_config)+1] .= 1
+        raw = transpose(raw)
```

Comment:
should copy here? since `transpose` is lazy.

Reply:
and `state` will always return a `transpose`ed version so `state(r) * mat` still works in YaoBlocks? and add a new function `storage` to return the raw storage?

Reply:
we need Tranpose wrapper here to create a effective row major array.